### ISSUE TITLE
Enhanced Scheduling with Calendar Integration

### ISSUE THREAD

**Issue by matbrgz** [ffplayout/ffplayout#469]
**Overview:**
Introduce a new feature to enhance the scheduling capabilities by integrating a calendar for a more user-friendly and intuitive program scheduling experience.

**Feature Details:**
Implement a calendar view within the scheduling module that allows users to:

1. **Drag-and-Drop Scheduling:** Enable users to easily drag and drop programs onto specific dates and times, streamlining the scheduling process.

2. **Intuitive Date Navigation:** Provide a user-friendly calendar interface for seamless navigation between different dates and weeks.

3. **Visual Representation:** Display a visual representation of the scheduled programs, allowing users to get a quick overview of the daily or weekly schedule.

**Screenshots for Reference:**
![image](https://github.com/ffplayout/ffplayout/assets/23284276/8ffccaaa-4abe-4e61-bc35-74813eb4bd35)
![image](https://github.com/ffplayout/ffplayout/assets/23284276/776ea8e8-736d-46ad-bb8d-a6ecb20bc6a5)

**Benefits of Enhanced Scheduling:**
1. **Improved Usability:** The drag-and-drop functionality simplifies the scheduling process, reducing the time and effort required to manage the programming calendar.

2. **Intuitive Navigation:** Users can easily explore and manage schedules through a familiar calendar interface, enhancing overall user experience.

3. **Visual Clarity:** A visual representation of scheduled programs provides clarity and helps users quickly identify gaps or conflicts in the schedule.

**Proposed Action:**
1. **Design Mockups:** Collaborate with the design team to create detailed design mockups for the calendar integration.

2. **Development Plan:** Outline a development plan for implementing the new scheduling features, including any backend changes required.

3. **User Feedback:** After development, gather feedback from users through beta testing to ensure the feature aligns with user expectations.

4. **Documentation:** Update documentation to guide users on utilizing the new scheduling capabilities effectively.

**Note:**
This feature aims to simplify the scheduling process and improve the overall user experience for managing program schedules.

**Comment by JunioCalu** [comment]
Could you take charge of doing this @matbrgz?

**Comment by matbrgz** [comment]
> Could you take charge of doing this @matbrgz?

I'm able to do that. But my major interested is migrating the entire frontend to React, I can help archive this too.

**Comment by JunioCalu** [comment]
> I'm able to do that. But my major interested is migrating the entire frontend to React, I can help archive this too.

Marvelous, so all that's left is for @jb-alvarado to give you the green light to start.

**Comment by jb-alvarado** [comment]
I understand that there is a need for it, and I can imagine doing some parts of it, but right now I don't see a chance for me to do the main tasks. I do not have that much capacity for it.

For my work, I programmed a scheduling solution, but it is highly customized to our needs. To create something where everyone is happy with it, I found very difficult. Not to mention all the feature requests and maintenance work that will come in the future.

Because we use our own separate scheduling solution, fflayout was never intended to be a full solution for everything. It was made for basic playout tasks, nothing more.

My recommendation would be to leave ffplayout/ffpapi/frontend as it is for now, with some refinements. Then create a separate backend and frontend for scheduling. With React I would be out for frontend work, but I could imagine to do things on backend side. Anyway I could maybe assist with thoughts because of my experience I have already.
